### ISSUE TITLE
Authentication bypassed if the site is internationalised

### ISSUE THREAD

**Issue by clementprdhomme** [labd/nextjs-basic-auth-middleware#29]
Hi,

I've been testing your middleware and I realised that if you define some locales inside `next.config.js`, then the authentication is automatically bypassed. If the user clicks “Cancel” multiple times, they may even see the page's HTML as reported in https://github.com/labd/nextjs-basic-auth-middleware/issues/27 before 2.0.0.

**Reproduction steps:**

1. Clone the repository: `git clone git@github.com:labd/nextjs-basic-auth-middleware.git`
2. Navigate to `nextjs-basic-auth-middleware/examples`
3. Install the dependencies: `yarn`
4. Add some locales to `next.config.js`:
```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  i18n: {
    locales: ['en', 'es'],
    defaultLocale: 'en',
  },
  reactStrictMode: true,
  swcMinify: true,
}

module.exports = nextConfig
```
5. Run the development server: `yarn dev`
6. Open the page: http://localhost:3000/

Even before clicking on “Cancel”, inside the “Network” developer panel, you will see that the page's HTML is returned to the browser.

**Comment by bramski** [comment]
Was there a workaround or this simply doesn't work?

**Comment by blurrah** [comment]
Quickly checked it and it fails this, I do get it to work after doing a reinstall of the package so I'll see if I can get it fixed soon. Does manually adding the locale paths to the middleware matcher work in the meantime?

**Comment by blurrah** [comment]
Bit late but it should be fixed in v3.0.1. Do note that it needs Next >=13.1 for this to work.
I want to extend tests with some e2e tests for these kinds of use cases to prevent regressions here.

Thanks for reporting :)
